src/typing_extensions.py

```diff
@@ -3531,11 +3531,24 @@ class TypeAliasType:
         def __init__(self, name: str, value, *, type_params=()):
             if not isinstance(name, str):
                 raise TypeError("TypeAliasType name must be a string")
+            if not isinstance(type_params, tuple):
+                raise TypeError("type_params must be a tuple")
             self.__value__ = value
             self.__type_params__ = type_params
 
+            default_value_encountered = False
             parameters = []
             for type_param in type_params:
+                if not isinstance(type_param, (TypeVar, TypeVarTuple, ParamSpec)):
```

Comment:
This needs to accept both the typing and typing_extensions versions of these classes, which are different on some versions. Test this.